ext/dapr-ext-workflow/dapr/ext/workflow/dapr_workflow_context.py

```diff
@@ -67,3 +68,78 @@ def wait_for_external_event(self, name: str) -> task.Task:
 
     def continue_as_new(self, new_input: Any, *, save_events: bool = False) -> None:
         self.__obj.continue_as_new(new_input, save_events=save_events)
+
+
+class CompositeTask(task.Task[T]):
+    """A task that is composed of other tasks."""
+    _tasks: List[task.Task]
+
+    def __init__(self, tasks: List[task.Task]):
+        super().__init__()
+        self._tasks = tasks
+        self._completed_tasks = 0
+        self._failed_tasks = 0
+        for _task in tasks:
+            _task._parent = self  # type: ignore
+            if _task.is_complete:
+                self.on_child_completed(_task)
+
+    def get_tasks(self) -> List[task.Task]:
+        return self._tasks
+
+    @abstractmethod
+    def on_child_completed(self, task: task.Task[T]):
+        pass
+
+
+class WhenAllTask(CompositeTask[List[T]]):
```

Comment:
This works, but I wonder if we're creating more work than necessary here? Could we just have `when_all` and `when_any` internally call the Durable Task versions of these same functions and return the resulting tasks directly to the user? I'm not sure if there's a lot of value in wrapping the Task return types from the durable SDK like we do for other APIs.

Alternatively, would it make sense to just have this `WhenAllTask` derive directly from the Durable Task `WhenAllTask` without reimplementing its logic?

Reply:
I have simply wrapped `when_all` and `when_any` from DTF-python.
I haven't wrapped return types - that doesn't sound useful.